### PR TITLE
java8 adjustments

### PR DESCRIPTION
**Pull request by patriknw**
got everything working in akka with these adjustments, and using `-Xdoclint:none`

there is one test failure that we can discuss tomorrow

### CODE REVIEW

**Comment by patriknw** [comment]
cc @rkuhn

**Comment by patriknw** [comment]
The test failure is on this line: https://github.com/typesafehub/genjavadoc/blob/master/javaOut/src/test/scala/scala/javadoc/SignatureSpec.scala#L64

It can be solved by adding `0 -> 1` in the `accProtLvl` Map, but I'm not sure if that is the correct solution.

File: plugin/src/main/scala/com/typesafe/genjavadoc/AST.scala [hunk 1]
```diff
@@ -96,7 +96,7 @@ trait AST { this: TransformCake ⇒
   def access(m: Modifiers, topLevel: Boolean): String = {
     if (m.isPublic) "public"
     else if (m.isProtected && !topLevel) "protected"
-    else if (m.isPrivate && !topLevel) "private"
+    else if (m.isPrivate && !topLevel) "" //"private"
```
Comment (patriknw) [inline plugin/src/main/scala/com/typesafe/genjavadoc/AST.scala]: the reason for changing this to package private is to avoid things like `static private class` and `private interface`

File: plugin/src/main/scala/com/typesafe/genjavadoc/BaseComments.scala [hunk 2]
```diff
@@ -58,7 +60,7 @@ trait BaseComments { this: TransformCake ⇒
       new Comment(pos, l2.reverse map htmlEntity)
     }
     private def preLine(line: String): String =
-      line.replace("@", "&#64;").replace("<", "&lt;").replace(">", "&gt;")
+      line.replace("@", "&#64;").replace("<", "&lt;").replace(">", "&gt;").replace("->", "-&gt;")
```
Comment (2m) [inline plugin/src/main/scala/com/typesafe/genjavadoc/BaseComments.scala]: Why is this needed? Isn't `.replace(">", "&gt;")` enough?

File: project/Build.scala [hunk 3]
```diff
@@ -16,7 +16,7 @@ object B extends Build {
   override lazy val settings = super.settings ++ Seq(
     organization := "com.typesafe.genjavadoc",
     version := "0.9-SNAPSHOT",
-    scalaVersion := "2.10.4",
+    scalaVersion := "2.11.5",
```
Comment (2m) [inline project/Build.scala]: Latest is `2.11.6`

File: plugin/src/main/scala/com/typesafe/genjavadoc/AST.scala (hunk 1 above)
Comment (2m) [inline plugin/src/main/scala/com/typesafe/genjavadoc/AST.scala]: Can it be more specific, like: `if (m.isInterface || m.hasStaticFlag) "" else "private"` ?

File: plugin/src/main/scala/com/typesafe/genjavadoc/BaseComments.scala (hunk 2 above)
Comment (patriknw) [inline plugin/src/main/scala/com/typesafe/genjavadoc/BaseComments.scala]: Good point, my addition is obviously not used, I think the problem was in java code in akka, so I changed that instead. I will remove this.

File: project/Build.scala (hunk 3 above)
Comment (patriknw) [inline project/Build.scala]: Yes, it was just that we had not change in akka yet.
How is the branching working here? Should master be on 2.11 or 2.10?

Comment (2m) [inline project/Build.scala]: I think `master` can be on `2.11` since there is no need for additional commits when cross releasing as `+publishSigned` should do all the work.

Also it can be on latest scala, because `crossScalaVersions` contains all scala versions. I would even change this line to read `crossScalaVersions.value.last` to remove duplication.

**Comment by patriknw** [comment]
I turns out that none of these changes are actually needed for making akka pass with java8. 
I had to do some changes to akka itself: https://github.com/akka/akka/commit/1d6ed6f1b4969a143862959f73559615460ed70d

I used `-Xdoclint:none`:

    javacOptions in doc ++= Seq("-encoding", "UTF-8", "-source", "1.8", "-Xdoclint:none"),

and then it is fine with `clean javaunidoc:doc`, but additional `javaunidoc:doc` without clean reports 100 errors, so I think there is something fishy with the `-Xdoclint:none` setting. Perhaps a bug in unidoc or sbt? Anyway, we always use clean genjavadoc.

**Comment by rkuhn** [comment]
Thanks Patrik, I have incorporate these into #50 which contains a whole lot of more fixes. Not yet to be merged, needs some more validation with Akka’s Java 8 port.
